moveit_planners/chomp/chomp_motion_planner/src/chomp_planner.cpp

```diff
@@ -156,16 +156,78 @@ bool ChompPlanner::solve(const planning_scene::PlanningSceneConstPtr& planning_s
 
   ros::WallTime create_time = ros::WallTime::now();
 
-  ChompOptimizer optimizer(&trajectory, planning_scene, req.group_name, &params, start_state);
-  if (!optimizer.isInitialized())
+  int replan_count = 0;
+  bool replan_flag = false;
+  double org_learning_rate = 0.04, org_ridge_factor = 0.0, org_planning_time_limit = 10;
+  int org_max_iterations = 200;
+
+  // storing the initial chomp parameters values
+  org_learning_rate = params.learning_rate_;
+  org_ridge_factor = params.ridge_factor_;
+  org_planning_time_limit = params.planning_time_limit_;
+  org_max_iterations = params.max_iterations_;
+
+  ChompOptimizer* optimizer;
+
+  // create a non_const_params variable which stores the non constant version of the const params variable
+  ChompParameters non_constant = non_constant.getNonConstantParams(params);
```

Comment:
better variable name - how about ``params_nonconst``

Reply:
sure, `params_nonconst` it is, done